gsa/src/web/pages/targets/component.js

```diff
@@ -15,406 +15,479 @@
+  const [port, setPort] = useState();
+  const [reverse_lookup_only, setReverseLookupOnly] = useState();
+  const [reverse_lookup_unify, setReverseLookupUnify] = useState();
+  const [target_source, setTargetSource] = useState();
+  const [target_exclude_source, setTargetExcludeSource] = useState();
+  const [targetTitle, setTargetTitle] = useState();
+  const [smb_credential_id, setSmbCredentialId] = useState();
+  const [ssh_credential_id, setSshCredentialId] = useState();
+  const [port_list_id, setPortListId] = useState();
+
+  // eslint-disable-next-line no-unused-vars
+  const [initial, setInitial] = useState({});
+
+  const [credentials, setCredentials] = useState();
+  const [portLists, setPortLists] = useState();
+  const [snmp_credential_id, setSnmpCredentialId] = useState();
+  const [portListsTitle, setPortListsTitle] = useState();
+
+  const openCredentialsDialog = ({id_field, types, title}) => {
+    setIdField(id_field);
+
+    setCredentialsDialogVisible(true);
+    setCredentialTypes(types);
+    setCredentialsTitle(title);
+
+    handleInteraction();
+  };
+
+  const closeCredentialsDialog = () => {
+    setCredentialsDialogVisible(false);
+  };
+
+  const handleCloseCredentialsDialog = () => {
+    closeCredentialsDialog();
+    handleInteraction();
+  };
+
+  const openTargetDialog = (entity, initial = {}) => {
     if (isDefined(entity)) {
-      this.setState({
-        targetDialogVisible: true,
-        id: entity.id,
-        alive_tests: entity.alive_tests,
-        comment: entity.comment,
-        esxi_credential_id: id_or__(entity.esxi_credential),
-        exclude_hosts: isDefined(entity.exclude_hosts)
-          ? entity.exclude_hosts.join(', ')
-          : '',
-        hosts: entity.hosts.join(', '),
-        in_use: entity.isInUse(),
-        name: entity.name,
-        port: isDefined(entity.ssh_credential)
-          ? entity.ssh_credential.port
-          : '22',
-        reverse_lookup_only: entity.reverse_lookup_only,
-        reverse_lookup_unify: entity.reverse_lookup_unify,
-        target_source: 'manual',
-        target_exclude_source: 'manual',
-        target_title: _('Edit Target {{name}}', entity),
-      });
+      setTargetDialogVisible(true);
+      setId(entity.id);
+      setAliveTests(entity.alive_tests);
+      setComment(entity.comment);
+      setEsxiCredentialId(id_or__(entity.esxi_credential));
+      setExcludeHosts(
+        isDefined(entity.exclude_hosts) ? entity.exclude_hosts.join(', ') : '',
+      );
+      setHosts(entity.hosts.join(', '));
+      setInUse(entity.isInUse());
+      setName(entity.name);
+      setPort(
+        isDefined(entity.ssh_credential) ? entity.ssh_credential.port : '22',
+      );
+      setReverseLookupOnly(entity.reverse_lookup_only);
+      setReverseLookupUnify(entity.reverse_lookup_unify);
+      setTargetSource('manual');
+      setTargetExcludeSource('manual');
+      setTargetTitle(_('Edit Target {{name}}', entity));
```

Comment:
Sadly putting every dialog state value into an own use state variable has a serious side effect. Every call to a setXyz function will cause a single re-render. Therefore either [useReducer](https://reactjs.org/docs/hooks-reference.html#usereducer) or a single state object must be used. E.g. i've used something like
```js
const [dialogState, setDialogState] = useState({foo: initialvalue, bar: initialvalue});

....
openDialog = () => {
setDialogState(state => {...state, xyzValue: 1234, foo: '345});
```
}